src/NuGet.Core/NuGet.ProjectModel/JsonPackageSpecReader.cs

```diff
@@ -953,7 +953,7 @@ private static void ReadMSBuildMetadata(JsonTextReader jsonReader, PackageSpec p
             RestoreAuditProperties auditProperties = null;
             bool useMacros = MSBuildStringUtility.IsTrue(environmentVariableReader.GetEnvironmentVariable(MacroStringsUtility.NUGET_ENABLE_EXPERIMENTAL_MACROS));
             var userSettingsDirectory = NuGetEnvironment.GetFolderPath(NuGetFolderPath.UserSettingsDirectory);
-            bool usingMicrosoftNetSdk = true;
```

Comment:
Just a quick clarification. 

I wasn't saying the reading for this property shouldn't default to true, I was merely saying the type shouldn't have it.

Reply:
We have now set the default value of the property to false. On the other hand, the package spec writer updates the dgspec file only when the `usingMicrosoftNetSdk` value is false. Consequently, tests creating a package spec for other purposes encounter errors because their expected JSON lacks `usingMicrosoftNetSdk`. To address this, I modified the writer to only update if the property is true or has been manually changed from the default false to true. Similarly, I updated the reader to adopt this pattern: default to false and only update if there has been a change in the package spec JSON. This approach makes the code intuitive and eliminates the need to update tests that utilize packagespec. Essentially, while we previously defaulted to true, we now default to false.

Reply:
My feedback may be idealistic or non-pragmatic, but I don't think that "a lot of tests need to be changed" is a good reason to make "worse" product changes. If tests are hard to update, then maybe the tests have tech debt that can be improved. or maybe this feature just naturally has a large impact and the work needs to be done. To me, what's more important is if the product change is good, tests aside, for the customer.

Going back to the original intent, most projects are now SDK style, and non-SDK style projects are no longer getting significant improvements, so over time we expect a larger percentage of projects to be SDK style. Maybe one day it'll reach 100%, but it'll never be 0%. The perf impact of avoiding writing `usingMicrosoftNetSdk` to the PackageSpec and the perf impact of having to read & parse it might be small, but restore is our most critical perf path, and some customers have 1000+ projects in a single restore.   I'm not saying we must micro-optimize these tiny perf improvements, but I am saying we must consider them.

Code [like this](https://github.com/dotnet/aspnetcore/blob/main/src/Shared/HttpSys/RequestProcessing/RequestHeaders.Generated.tt) isn't written because it's easy to maintain, but because the perf requirements demand it. So the question to the team is if the reduction in maintainability of having "no value in DGSpec means this value is true" is worth the perf gain. I have an opinion, but I'm just one person in the team.  But I don't think that the tests needing a lot of effort to fix is a good justification, it should be framed as customer experience, and engineering effort applies if the customer impact is the same.

Reply:
I agree with both of you. I think we should prioritize performance. I will revert the code.